inc/datahandlers/warnings.php

```diff
@@ -737,5 +738,22 @@ function update_warning()
 		return $this->write_warning_data;
 	}
 
-}
+	/**
+	 * Acknowledges a warning
+	 *
+	 */
+	function acknowledge() : void
```

Comment:
Make sure to add a hook he, though the best location would be better out figured after you update the logic.